### PR TITLE
Initial support of Mysterious Combination Book Bundle (Item: 23817)

### PR DESCRIPTION
**Pull request by llchrisll**
- Every "Combination Book"
- Missing items and their effects
- Updated MvP's HP/EXP and Drop Rates according to divine-pride.net for the MvP Adjustment update
- Added unofficial conversation to the Rough Ore Refiner (Christopher - geffen_in) for Shadowdecon and Zelunium

Note: Drop chance of Mysterious Combination Book Bundle and new materials are customized, since there is no accurate information about it. At least I didn't find anything relevant.

Signed-off-by: llchrisll <h0und3y3@gmail.com>

* **Server Mode**: 
Renewal
* **Description of Pull Request**: 
I don't guarantee the correctness of the item scripts at all, I based it on my understanding of the item descriptions in Zackdreaver's Translation Project. Also it's basically my first time doing this stuff.
Also for testing purposes, here is the itemInfo.lua entries:
[itemInfo.zip](https://github.com/rathena/rathena/files/3545042/itemInfo.zip)
This is only a part of 16th Anniversary update:
Source: https://www.divine-pride.net/forum/index.php?/topic/3280-16th-anniversary-update-note-part-1/&do=findComment&comment=4754
You still need the "Green MvP Effect"!

### CODE REVIEW

**Comment by cahya1992** [comment]
it should be something related with feature laphine dukkdakk but rathena didn't support this feature

https://www.youtube.com/watch?v=Lov9M3XDdb4

File: db/re/mob_db.txt [hunk 1]
```diff
@@ -115,10 +115,10 @@
 1109,DEVIRUCHI,Deviruchi,Deviruchi,93,8912,1,1806,2166,1,477,182,72,16,61,49,30,85,99,5,10,12,0,6,27,0x2003695,150,980,600,384,0,0,0,0,0,0,0,1038,5335,1039,400,984,2,1458,2,1009,5,912,1500,756,154,0,0,0,0,4122,1
 1110,DOKEBI,Dokebi,Dokebi,68,2820,1,675,759,1,333,30,85,20,52,56,35,20,60,25,10,12,0,6,27,0x2000091,250,1156,456,384,0,0,0,0,0,0,0,1021,9000,757,150,1517,2,1613,1,969,1,1501,300,1005,5,0,0,0,0,4098,1
 1111,DRAINLIAR,Drainliar,Drainliar,47,1162,1,389,438,1,100,42,50,15,35,34,24,22,50,20,10,12,0,2,47,0x3095,250,1276,576,384,0,0,0,0,0,0,0,1011,60,913,3000,725,20,507,1000,7006,5500,7006,1500,756,40,0,0,0,0,4069,1
-1112,DRAKE,Drake,Drake,91,804500,1,515160,417960,1,2519,1403,279,135,121,103,114,107,135,71,10,12,1,1,29,0x6283695,400,620,420,360,257580,504,5000,719,500,0,0,1127,600,1125,950,1135,150,1128,400,5019,350,985,3200,1189,100,0,0,0,0,4137,1
+1112,DRAKE,Drake,Drake,91,804500,1,309.096,208980,1,2519,1403,279,135,121,103,114,107,135,71,10,12,1,1,29,0x6283695,400,620,420,360,154548,23817,2500,719,500,0,0,1127,300,1125,475,1135,75,1128,200,5019,175,985,1600,1189,50,0,0,0,0,4137,1
```
[the remark this answers is not in the document]
Comment (llchrisll) [reply db/re/mob_db.txt]: Oh might have missed that, will fix it after work ty :).

**Comment by cahya1992** [comment]
In official when we doesn't have all requirements item. the scroll didnt consume or dissapear

**Comment by llchrisll** [comment]
> In official when we doesn't have all requirements item. the scroll didnt consume or dissapear

Hmm, didn't know that, question for me now is how to prevent the consumption, does "end;" work? alternate solution: generate scroll again.

**Comment by cahya1992** [comment]
> > In official when we doesn't have all requirements item. the scroll didnt consume or dissapear
> 
> Hmm, didn't know that, question for me now is how to prevent the consumption, does "end;" work? alternate solution: generate scroll again.


Coz its should new interface from feature lapine dukkdakk 
something like this http://prntscr.com/oz1uky

you can added this code in your itemdb
` else { getitem 23819,1; callfunc "F_Requireditem"; }`

 function	script	F_Requireditem	{

		dispbottom "System: You didn't have enough items to make this item!.",0xe63900;
		end;
	
}

**Comment by llchrisll** [comment]
In my opinion the dukkdakk box is something completely different, as the required items for the recipes are different.
Also I knew about the getitem way, but was wondering if there was a different solution to prevent the consumption to begin with.
But I wouldn't use an callfunc for just an dispbottom....
Could have made an function for the whole script part then.
I mean I could but why xD? for 4 lines of code? nope
Well I pretty much repeat the code, but with different values.
``function	script	F_Combination_Book	{
// getarg(0) = Material 1
// getarg(1) = Amount
// getarg(2) = Material 2
// getarg(3) = Amount
// getarg(4) = Result Item ID
// getarg(5) = Combination Book Item ID
	
	if(getarg(0) >= getarg(1) && getarg(2) >= getarg(3)) {
		delitem getarg(0),getarg(1);
		delitem getarg(2),getarg(3);
		getitem getarg(4),1;
	} else {
		getitem getarg(5),1;
		dispbottom "System: You don't have the required materials for this recipe.",0xe63900;
		end;
	}
}``
Something like that I would have done otherwise.
And in Item Script for Item ID 23819 as example:
``callfunc "F_Combination_Book",7510,150,25731,10,5548,23819;``

**Comment by llchrisll** [comment]
> > > In official when we doesn't have all requirements item. the scroll didnt consume or dissapear
> > 
> > 
> > Hmm, didn't know that, question for me now is how to prevent the consumption, does "end;" work? alternate solution: generate scroll again.
> 
> Coz its should new interface from feature lapine dukkdakk
> something like this http://prntscr.com/oz1uky
> 
> you can added this code in your itemdb
> ` else { getitem 23819,1; callfunc "F_Requireditem"; }`
> 
> function	script	F_Requireditem	{
> 
> ```
> 	dispbottom "System: You didn't have enough items to make this item!.",0xe63900;
> 	end;
> ```
> 
> }

Added in 867ae8c37d0366dd2ed5a241659f0d3afb36d258

**Comment by llchrisll** [comment]
Gonna close this PR on my own accord, as I deleted my fork for it already. So I can't make any changes if they would be required. Also since I mostly did those changes without aquiring detailed official informations, so it's not really accurate ofc.

**Pull request closed without merging**